src/peergos/server/corenode/JDBCCoreNode.java

```diff
@@ -603,26 +163,53 @@ public boolean updateChain(String username,
         return CompletableFuture.completedFuture(resp.serialize());
     }
 
-    public CompletableFuture<Boolean> setPointer(PublicKeyHash owner, PublicKeyHash writerHash, byte[] writingKeySignedHash) {
-        MetadataBlob blob = new MetadataBlob(writerHash.serialize(), writingKeySignedHash);
-        return CompletableFuture.completedFuture(blob.insert());
+    public CompletableFuture<Boolean> setPointer(PublicKeyHash writingKey, Optional<byte[]> existingCas, byte[] newCas) {
+        if (existingCas.isPresent()) {
+            try (PreparedStatement insert = conn.prepareStatement("UPDATE metadatablobs SET hash=? WHERE writingkey = ? AND hash = ?")) {
```

Comment:
extract out query string

Reply:
Where would you put it, and why?

Reply:
convention mainly. Always had the habit of making static final class vars. In a CRUD app having all sql in one place can make subsequent changes to field names less error prone. not a concern here.

Reply:
Done. That's a good argument.